modules/explorer/info_test.go

```diff
@@ -72,3 +74,190 @@ func TestBlockFacts(t *testing.T) {
+		WindowStart:        windowStart,
+		WindowEnd:          windowEnd,
+		Payout:             payout,
+		ValidProofOutputs:  []types.SiacoinOutput{{Value: types.PostTax(et.cs.Height(), payout)}},
+		MissedProofOutputs: []types.SiacoinOutput{{Value: types.PostTax(et.cs.Height(), payout)}},
+		UnlockHash:         types.UnlockConditions{}.UnlockHash(),
+	}
+
+	fcIndex := builder.AddFileContract(fc)
+	tSet, err := builder.Sign(true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = et.tpool.AcceptTransactionSet(tSet)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Mine until contract payout is in consensus
+	for i := et.cs.Height(); i < windowEnd+types.MaturityDelay; i++ {
+		_, err := et.miner.AddBlock()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	ti := len(tSet) - 1
+	fcid := tSet[ti].FileContractID(fcIndex)
+	txns := et.explorer.FileContractID(fcid)
+	if len(txns) == 0 {
+		t.Error("Filecontract ID does not appear in blockchain")
+	}
+
+	outputs, err := et.explorer.FileContractPayouts(fcid)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Check if MissedProofOutputs were added to spendable outputs
+	if len(outputs) != len(fc.MissedProofOutputs) {
+		t.Error("Incorrect number of outputs returned")
+		t.Error("Expecting -> ", len(fc.MissedProofOutputs))
+		t.Error("But was -> ", len(outputs))
+	}
+}
+
+func TestFileContractsPayoutValidProof(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+
+	et, err := createExplorerTester(t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// COMPATv0.4.0 - Step the block height up past the hardfork amount. This
+	// code stops nondeterministic failures when producing storage proofs that
+	// is related to buggy old code.
+	for et.cs.Height() <= 10 {
+		_, err := et.miner.AddBlock()
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	// Create a file (as a bytes.Buffer) that will be used for the file
+	// contract.
+	filesize := uint64(4e3)
+	file := fastrand.Bytes(int(filesize))
+	merkleRoot := crypto.MerkleRoot(file)
+
+	// Create a funded file contract
+	payout := types.NewCurrency64(400e6)
+	fc := types.FileContract{
+		FileSize:           filesize,
+		FileMerkleRoot:     merkleRoot,
+		WindowStart:        et.cs.Height() + 1,
+		WindowEnd:          et.cs.Height() + 2,
+		Payout:             payout,
+		ValidProofOutputs:  []types.SiacoinOutput{{Value: types.PostTax(et.cs.Height(), payout)}},
+		MissedProofOutputs: []types.SiacoinOutput{{Value: types.PostTax(et.cs.Height(), payout)}},
+	}
+
+	// Submit a transaction with the file contract.
+	//oldSiafundPool := cst.cs.dbGetSiafundPool()
+	builder := et.wallet.StartTransaction()
+	err = builder.FundSiacoins(payout)
+	if err != nil {
+		panic(err)
+	}
+
+	fcIndex := builder.AddFileContract(fc)
+	tSet, err := builder.Sign(true)
+	if err != nil {
+		panic(err)
+	}
+
+	err = et.tpool.AcceptTransactionSet(tSet)
+	if err != nil {
+		panic(err)
+	}
+	_, err = et.miner.AddBlock()
+	if err != nil {
+		panic(err)
```

Comment:
Panicking is pretty much a last resort. If it's present in other tests, it's because we were less careful back then. :P sometimes it can be convenient if you don't have access to the testing object.